src/storage/buffer/buffer_pool.cpp

```diff
@@ -1,12 +1,41 @@
 #include "duckdb/storage/buffer/buffer_pool.hpp"
 
+#include "duckdb/common/chrono.hpp"
 #include "duckdb/common/exception.hpp"
 #include "duckdb/parallel/concurrentqueue.hpp"
 #include "duckdb/storage/temporary_memory_manager.hpp"
-#include "duckdb/common/chrono.hpp"
+#include <cstddef>
+#include <cstdint>
+
+#if defined(_WIN32)
+#include <windows.h>
+#elif defined(__GNUC__)
+#include <sched.h>
+#include <unistd.h>
+#endif
 
 namespace duckdb {
 
+// this code comes from jemalloc
+static int GetCurrentCPU() {
```

Comment:
We try to avoid the `int` data type - can we turn this into either an `int32_t`, `int64_t` or `idx_t`?